shared/src/vyxal/Elements.scala

```diff
@@ -2263,6 +2313,159 @@ object Elements:
             "Maximum By: First argument should be a function"
           )
     },
+    addDirect(
+      "#|apply-to-register",
+      "[Internal Use] Apply to Register (Element Form)",
+      List(),
+      None,
+      "*a, f -> f applied to the register. Use the modifier instead.",
+    ) { ctx ?=>
+      ctx.pop() match
+        case f: VFun =>
+          ctx.push(ctx.globals.register)
+          ctx.push(Interpreter.executeFn(f)(using ctx.makeChild()))
+          ctx.globals.register = ctx.pop()
+        case _ => throw IllegalArgumentException(
+            "Apply to Register: First argument should be a function"
+          )
+
+    },
+    addDirect(
+      "#|dip",
+      "[Internal Use] Dip (Element Form)",
+      List(),
+      None,
+      "*a, f -> f applied to a with a pushed back. Use the modifier instead.",
+    ) { ctx ?=>
+      val f = ctx.pop()
+      val top = ctx.pop()
+      f match
+        case fun: VFun =>
+          Interpreter.executeFn(fun)(using ctx.makeChild())
+          ctx.push(top)
+        case _ => throw IllegalArgumentException(
+            "Dip: First argument should be a function"
+          )
+    },
+    addDirect(
+      "#|invar",
+      "[Internal Use] Invariant (Element Form)",
+      List(),
+      None,
+      "*a, f -> Use the ᵞ modifier instead.",
+    ) { ctx ?=>
+      val f = ctx.pop()
+      val copy = ctx.peek
+      f match
+        case fun: VFun =>
+          val result = Interpreter.executeFn(fun)(using ctx.makeChild())
+          ctx.push(result === copy)
+        case _ => throw IllegalArgumentException(
+            "Invariant: First argument should be a function"
+          )
+    },
+    addDirect(
+      "#|vscan",
+      "[Internal Use] Vectorised Scan (Element Form)",
+      List(),
+      None,
+      "*a, f -> scanl each column. Use the modifier instead.",
+    ) { ctx ?=>
+      val f = ctx.pop()
+      val arg = ListHelpers
+        .makeIterable(ctx.pop())
+        .map(x => ListHelpers.makeIterable(x))
+      f match
+        case fun: VFun => ctx.push(
+            VList.from(
```

Comment:
Indexing can be slow on big linked lists. I think we could try something like

```scala
LazyList.unfold(arg) { rows =>
  val rem = rows.filter(_.nonEmpty)
  Option.when(rem.nonEmpty) {
    val col = rem.map(_.head)
    val nextRows = rem.map(_.tail)
    (MiscHelpers.scanl(col, fun), nextRows)
  }
}
```

This is completely untested, though, and I have a splitting headache right now so don't blame me if it doesn't work